### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by nyuuyn**
adds base junit class for testing

#### Short Description
(Re)enable junit testing within the runtime as we now have a pure maven project

#### Proposed Changes
  * Add base junit classes for testing low and high level functionality
  * Add at least a Test with a complete csar

### CODE REVIEW

File: org.opentosca.bus/org.opentosca.bus.application.api.jsonhttp/src/main/java/org/opentosca/bus/application/api/jsonhttp/route/Route.java [hunk 1]
```diff
@@ -46,25 +46,25 @@ public void configure() throws Exception {
...
-                                                                                 .removeHeaders("*");
+            .to(Route.TO_APP_BUS_ENDPOINT)
+            .process(isFinishedResponseProcessor)
+            .removeHeaders("*");
 
         // getResult route
         from("rest:" + Route.BASE_ENDPOINT + Route.GET_RESULT_ENDPOINT
             + "?method=get").process(getResultRequestProcessor).to(Route.TO_APP_BUS_ENDPOINT)
-                            .process(getResultResponseProcessor).removeHeaders("*");
+            .process(getResultResponseProcessor).removeHeaders("*");
```
Comment (zimmerml) [inline org.opentosca.bus/org.opentosca.bus.application.api.jsonhttp/src/main/java/org/opentosca/bus/application/api/jsonhttp/route/Route.java:68]: Why the reformat? Was the code style not followed as specified [here](https://github.com/OpenTOSCA/container/tree/master/docs/codestyle)?

File: org.opentosca.bus/org.opentosca.bus.application.api.soaphttp/src/main/java/org/opentosca/bus/application/api/soaphttp/route/Route.java [hunk 2]
```diff
@@ -16,14 +16,14 @@
...
- * routed to processors or the application bus in order to handle the requests.
+ * The endpoint of the SOAP/HTTP-API is created here. Incoming requests will be un/marshalled, routed to processors or
+ * the application bus in order to handle the requests.
  *
  * @author Michael Zimmermann - zimmerml@studi.informatik.uni-stuttgart.de
  */
 public class Route extends RouteBuilder {
 
-    private final static String ENDPOINT = "http://0.0.0.0:8084/appBus";
+    private final static String ENDPOINT = "http://0.0.0.0:8082/appBus";
```
Comment (zimmerml) [inline org.opentosca.bus/org.opentosca.bus.application.api.soaphttp/src/main/java/org/opentosca/bus/application/api/soaphttp/route/Route.java:26]: Why were the ports changed?

**Review by zimmerml — APPROVED** [review]
lgtm

File: org.opentosca.bus/org.opentosca.bus.application.api.soaphttp/src/main/java/org/opentosca/bus/application/api/soaphttp/route/Route.java (hunk 2 above)
Comment (nyuuyn) [reply org.opentosca.bus/org.opentosca.bus.application.api.soaphttp/src/main/java/org/opentosca/bus/application/api/soaphttp/route/Route.java:26]: that port is used in the github actions runner by something else...sadly

File: org.opentosca.bus/org.opentosca.bus.application.api.jsonhttp/src/main/java/org/opentosca/bus/application/api/jsonhttp/route/Route.java (hunk 1 above)
Comment (nyuuyn) [reply org.opentosca.bus/org.opentosca.bus.application.api.jsonhttp/src/main/java/org/opentosca/bus/application/api/jsonhttp/route/Route.java:68]: The codestlye is the one defined within the intellij project files on master, maybe a PR which establishes codestyle in all IDE project files would help here

Comment (zimmerml) [reply org.opentosca.bus/org.opentosca.bus.application.api.jsonhttp/src/main/java/org/opentosca/bus/application/api/jsonhttp/route/Route.java:68]: Maybe the provided codestyle settings in the docs folder should be updated/deleted then.

**Review by wederbn — APPROVED** [review]

**Pull request merged**